src/Extensions/Diagnostics.HealthChecks/ServiceCollectionExtensions.cs

```diff
@@ -30,20 +31,47 @@ public static IHealthChecksBuilder AddServiceFabricHealthChecks(this IServiceCol
 				});
 
 			serviceCollection.TryAddSingleton<ObjectPoolProvider, DefaultObjectPoolProvider>();
+		}
 
-			return serviceCollection
-				.AddHealthCheckPublisher<ServiceFabricHealthCheckPublisher>()
-				.AddHealthChecks();
+		/// <summary>
+		/// Register publisher for processing health check results directly to replicas
+		/// </summary>
+		public static IHealthChecksBuilder AddServiceFabricHealthChecks(this IServiceCollection serviceCollection)
+		{
+			return serviceCollection.AddServiceFabricHealthChecks<ServiceFabricHealthCheckPublisher>();
+		}
+
+		/// <summary>
+		/// Register publisher for processing health check results directly to nodes using REST api
+		/// </summary>
+		public static IHealthChecksBuilder AddRestHealthChecks(this IServiceCollection serviceCollection)
+		{
+			serviceCollection.TryAddSingleton(
+				provider =>
+				{
+					return provider.GetRequiredService<IOptions<RestHealthCheckPublisherOptions>>().Value;
+				});
+			return serviceCollection.AddServiceFabricHealthChecks<RestHealthCheckPublisher>();
 		}
 
 		/// <summary>
 		/// Register publisher for processing health check results
 		/// </summary>
+		public static IHealthChecksBuilder AddServiceFabricHealthChecks<TPublisher>(this IServiceCollection serviceCollection)
+				where TPublisher : class, IHealthCheckPublisher
+		{
+			serviceCollection.ConfigureService();
```

Comment:
Why this line needed? If I'm not mistaken this call does not do anything.

Reply:
It calls configuration for HTTP communication:
```cs
private static void ConfigureService(this IServiceCollection serviceCollection)
		{
			serviceCollection
				.AddHttpClient(HttpEndpointHealthCheck.HttpClientLogicalName)
				.ConfigurePrimaryHttpMessageHandler(() => new HttpClientHandler
				{
					AllowAutoRedirect = false,
					Credentials = CredentialCache.DefaultCredentials,
					ServerCertificateCustomValidationCallback = (sender, x509Certificate, chain, errors) => true
				});

			serviceCollection.TryAddSingleton<ObjectPoolProvider, DefaultObjectPoolProvider>();
		}
```

It was used also before: [line](https://github.com/microsoft/Omex/blob/master/src/Extensions/Diagnostics.HealthChecks/ServiceCollectionExtensions.cs#L23)

Reply:
Or I have not realized that it's our method, we would need to think about better naming here, ex `AddPublisherDepenencies()`.